classes/publication/DAO.php

```diff
@@ -175,6 +177,12 @@ public function fromRow(object $row): Publication
             ->value('locale');
         $publication->setData('locale', $locale);
 
+        $citationDao = DAORegistry::getDAO('CitationDAO'); /** @var CitationDAO $citationDao */
```

Comment:
Hmm, this will potentially be expensive for operations that iterate through a list of publications -- e.g. the submission lists. We're going through the whole citation object lifecycle for every citation attached to each submission, just to get the `citationsRaws`, which in most cases likelihood won't be used.

It wouldn't completely eliminate the overhead, but it might be better to add a function to CitationDAO that just gets the set of citations glued together as a string rather than going through the whole object instantiation part.

Better still, for the few operations that still use `citations` and `citationsRaw`, they could be adapted to use the DAO directly rather than relying on the data being in the DataObject.

Reply:
Hi @asmecher, hmmm... yes...
But the citationsRaw is everywhere expected to be part of the publication object -- not only in smarty (this is minor), but also in the Form, API and Vue.js... So I would rather leave this as it is for now.
I can try to have a function in CitationDAO that would generate that citationsRaw string.
I can then see with Vitaly and Jarda, if we should change something -- for citations not to be part of the publication object -- when we work on structured citations.
OK?

Reply:
Hi @asmecher, hmmm.... another problem/question: if we concatenate it with SQL, the resulting string will be truncated to the `group_concat_max_len`, which is 1024 per default, which seems not to be that much for citations, no?

Reply:
Hi @asmecher, one more thing: The `citations` and `citationsRaw` are not part of submission list properties, so these do not influence the performance of it.
However it is maybe good to get raw citations with SQL and then implode them to a string -- instead of going through every object.
And, actually we should always get citation from the publication object -- then they are queried only once here and used everywhere else. Thus, I will now also change those other places that use $citationDao->getByPublicationId() over and over again.

Reply:
@bozana, I think we don't need to concatenate in SQL -- we can still do it in PHP. But it'll be a lot more efficient to just iterate through a result set:
```php
DB::table('citations')->where('publication_id', $publicationId)->implode(...);
```
...rather than going through the whole `Citation` / `CitationDAO` object lifecycle.